core/tee/tadb.c

```diff
@@ -670,17 +694,25 @@ TEE_Result tee_tadb_ta_read(struct tee_tadb_ta *ta, void *buf, size_t *len)
 			return res;
 	} else {
 		size_t num_bytes = 0;
+		size_t b_size = MIN(SIZE_4K, l - num_bytes);
```

Comment:
You zero out `num_bytes` at the line above. Better to leave only `MIN(SIZE_4k, l)`. It is easier to read.